.travis.yml

```diff
@@ -11,6 +11,9 @@ language: python
 python:
   - "2.7"
 
+cache:
+  - pip
+
 # command to install dependencies
 install:
   - sudo apt-get install graphviz
```

Comment:
@bhuvangupta008 
can we move all of these installs including the apt-get line into setup.py?

Reply:
the pip dependencies can. They should go in the extra_requires:dev section.

Reply:
although the python setup.py install command should change. See https://stackoverflow.com/questions/28509965/setuptools-development-requirements

Reply:
There is a really good writeup on [ubuntu forums](https://askubuntu.com/questions/95037/what-is-the-best-way-to-install-python-packages) by Mark Mikofski advocating the use of deb packages over pip packages. It makes a lot of sense.

The gist is simple: 
1. For python packages coming in via apt-get, all the OS-dependencies get installed. With pip there is no such guarantee. 
1. With deb packages there is a package signatures, but with pip there is none.
1. With deb packages, you can cleanly uninstall, but with pip setup.py there is no clean rollback.